File: PuntoC/Punto3D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PuntoC
{
    public class Punto3D:Punto2D
    {
        protected int _z;

        public int Z
        {
            get { return _z; }
            set { _z = value; }
        }
        public Punto3D()
        {
            _x = 0;
            _y= 0;
            _z = 0;
        }
        public Punto3D(int x, int y, int z) {
            _x = x;
            _y = y;
            _z = z;
        }
        public override string ToString()
        {
            return "("+_x+","+_y+","+_z+")";
        }
        public double Distancia(Punto3D p)
        {
            double resultado = 0;
            double xp = (_x - p.X) * (_x - p.X);
            double yp = (_y - p.Y) * (_y - p.Y);
            double zp = (_z - p.Z) * (_z - p.Z);
            resultado = Math.Sqrt(xp + yp+zp);
            return resultado;
        }
    }

}

File: PuntoVB/Punto3D.vb
Public Class Punto3D

    Inherits Punto2D

    Private z1 As Integer

    Public Sub New()
        x1 = 0
        y1 = 0
        z1 = 0
    End Sub
    Public Sub New(x As Integer, y As Integer, z As Integer)
        x1 = x
        y1 = y
        z1 = z
    End Sub

    Public Property Z() As Integer
        Get
            Return z1
        End Get

        Set(ByVal value As Integer)
            z1 = value
        End Set
    End Property

    Public Overloads Function Distancia(p As Punto3D) As Double
        Dim resultado As Double = 0
        Dim xp = (x1 - p.X) * (x1 - p.X)
        Dim yp = (y1 - p.Y) * (y1 - p.Y)
        Dim zp = (z1 - p.Z) * (z1 - p.Z)
        resultado = Math.Sqrt(xp + yp + zp)
        Return resultado
    End Function
    Public Overrides Function ToString() As String
        Return "(" + x1 + "," + y1 + "," + z1 + ")"
    End Function
End Class
